src/components/ResultsView.tsx

```diff
@@ -123,14 +123,15 @@ const ResultHeader = ({
 };
 
 const CellEmbed = ({ uid }: { uid: string }) => {
+  const isPage = !!getPageTitleByPageUid(uid) ? "page-embed" : "block-embed"
```

Comment:
Let's make sure we only call `getPageTitleByPageUid(uid)` once - we want to do Roam data lookups as infrequently as possible:
```typescript
const title = getPageTitleByPageUid(uid);
const isPage = !!title ? "page-embed" : "block-embed";
...
<h1 className="rm-title-display">{title}</h1>
```